extensions/rapidcap/src/create-card.tsx

```diff
@@ -0,0 +1,68 @@
+import { Form, Action, ActionPanel, useNavigation, getPreferenceValues, showToast, Toast } from "@raycast/api";
+import { FormValidation, useForm } from "@raycast/utils";
+import { useEffect, useState } from "react";
+import { Preferences, Card } from "./types";
+import fs from "fs";
+import validateTag from "./utils";
+
+function CreateCardAction() {
+  const preferences = getPreferenceValues<Preferences>();
+  const { pop } = useNavigation();
+  const [error, setError] = useState<Error>();
+
+  useEffect(() => {
+    if (error) {
+      showToast({
+        style: Toast.Style.Failure,
+        title: "Something went wrong",
+        message: error.message,
+      });
+    }
+  }, [error]);
+
+  const { handleSubmit, itemProps } = useForm<Card>({
+    onSubmit(values: { question: string; answer: string; tag: string }) {
+      fs.readFile(preferences.dataFile, "utf-8", (error, data) => {
+        if (error) {
+          setError(error);
+        }
+
+        const cards: Card[] = JSON.parse(data);
```

Comment:
This probably should be a try-catch, in case the data gets corrupted (which is possible, as it is intended for users to directly edit this file), or if the user has an empty file.

Reply:
Yeah that is a good point